hera_sim/sigchain.py

```diff
@@ -1086,12 +1086,15 @@ def build_coupling_matrix(
             else:
                 power_beam = uvbeam.copy()
                 power_beam.efield_to_power()
-                if power_beam.interpolation_function is None:
-                    power_beam.interpolation_function = pixel_interp
+                # if power_beam.interpolation_function is None:
+                #    power_beam.interpolation_function = pixel_interp
                 if power_beam.freq_interp_kind is None:
                     power_beam.freq_interp_kind = freq_interp
                 power_beam = power_beam.interp(
-                    freq_array=freqs * units.GHz.to("Hz"), new_object=True
+                    freq_array=freqs * units.GHz.to("Hz"),
+                    new_object=True,
+                    interpolation_function=pixel_interp,
+                    #                    freq_interp_func=freq_interp,
```

Comment:
Does the new API have a `freq_interp_func` parameter, or do we need to stick with what's on lines 1091-1092? If it's the latter, then let's remove this line; if it's the former, then let's remove lines 1091-1092 and uncomment this.

Reply:
I don't believe it does, which was confusing to me.